Sources/PointFree/Database.swift

```diff
@@ -2,88 +2,36 @@ import Either
 import Foundation
 import Prelude
 
-#if os(iOS)
-  enum PostgreSQL {
-    enum ConnInfo {
-      case basic(String, Int, String, String, String)
-    }
-    struct Database {
-      init(_ connInfo: ConnInfo) throws {
-      }
-      func makeConnection() throws -> Connection {
-        return Connection()
-      }
-    }
-    struct Connection {
-      func execute(_ query: String, _ representable: [NodeRepresentable] = []) throws -> Node {
-        return Node()
-      }
-    }
-    public typealias NodeRepresentable = Any
-    struct Node {
-      subscript(_: Int, _: String) -> Node? {
-        return nil
-      }
-      var int: Int?
-      var string: String?
-    }
+public struct Database {
+  var createSubscription: (StripeSubscription, User) -> EitherIO<Error, Prelude.Unit>
+  var createUser: (GitHub.UserEnvelope) -> EitherIO<Error, Prelude.Unit>
+  var fetchUser: (GitHub.AccessToken) -> EitherIO<Error, User?>
+  var migrate: () -> EitherIO<Error, Prelude.Unit>
+
+  static let `default` = Database(
```

Comment:
Using a `default` constructor instead of defaulting assignments to avoid accidentally calling live side effects in our test environment.

Reply:
could maybe call it `live` then? `Database.live` vs `Database.mock`?